### PR TITLE
SCMOD-11850: Optimized tests to use conditional wait statement

### PR DESCRIPTION
**Pull request by pintusingh135**
Replaced the Thread.sleep methods in caf-job-service automation test cases with a conditional wait method for a better execution time.

### CODE REVIEW

File: job-service-acceptance-tests/src/test/java/com/hpe/caf/jobservice/acceptance/JobServiceEndToEndIT.java [hunk 1]
```diff
@@ -1329,4 +1341,19 @@ public void run() {
...
         return timer;
     }
+
+    /**
+     * Desc: Waits till the job is Completed
+     * @param jobId
+     * @throws ApiException
+     * @throws InterruptedException
+     */
+    private void waitUntilJobCompletes(String jobId) throws ApiException, InterruptedException {
```
Comment (michael-bryson) [inline job-service-acceptance-tests/src/test/java/com/hpe/caf/jobservice/acceptance/JobServiceEndToEndIT.java:1353]: What happens if there is an issue with the job and it never completes, will this get stuck in the loop? Why not use a for loop to ensure you exit out after a certain number of tries.
```suggestion
    private void waitUntilJobCompletes(String jobId) throws ApiException, InterruptedException {
        for (int i = 0; i < 60; i++) {
            Job job = jobsApi.getJob(defaultPartitionId, jobId, jobCorrelationId);
            String currentJobStatus = job.getStatus();
            LOG.info("Job " + jobId + " current status: " + currentJobStatus);
            if (currentJobStatus.equalsIgnoreCase("completed")) {
                break;
            }
            else {
                Thread.sleep(500);
            }
        }
    }
```

File: job-service-acceptance-tests/src/test/java/com/hpe/caf/jobservice/acceptance/JobServiceEndToEndIT.java [hunk 2]
```diff
@@ -240,13 +240,15 @@ public void testTargetPipeForJobWithNoAndWithCompletedPrerequisiteJobs(final Str
             createJob(job2Id, true);
         }
 
-        Thread.sleep(1000); // Add short delay to allow previous jobs to complete
+        //Wait for the job to complete
```
Comment (michael-bryson) [inline job-service-acceptance-tests/src/test/java/com/hpe/caf/jobservice/acceptance/JobServiceEndToEndIT.java]: Should `waitUntilJobCompletes(job1Id);` be in this test also?

File: job-service-acceptance-tests/src/test/java/com/hpe/caf/jobservice/acceptance/JobServiceEndToEndIT.java [hunk 3]
```diff
@@ -350,13 +352,15 @@ public void testJobWithPrerequisiteJobsWhichHaveCompleted() throws Exception
...
```
Comment (michael-bryson) [inline job-service-acceptance-tests/src/test/java/com/hpe/caf/jobservice/acceptance/JobServiceEndToEndIT.java:356]: Should `waitUntilJobCompletes(job1Id);` be in this test also?

**Review by michael-bryson — CHANGES_REQUESTED** [review]

File: job-service-acceptance-tests/src/test/java/com/hpe/caf/jobservice/acceptance/JobServiceEndToEndIT.java [hunk 4]
```diff
@@ -1329,4 +1343,23 @@ public void run() {
...
+
+    /**
+     * Desc: Waits till the job is Completed
+     * @param jobId
+     * @throws ApiException
+     * @throws InterruptedException
+     */
+    private void waitUntilJobCompletes(String jobId) throws ApiException, InterruptedException {
+        long currentTime= System.currentTimeMillis();
+        long endTime = currentTime+5000;
```
Comment (michael-bryson) [inline job-service-acceptance-tests/src/test/java/com/hpe/caf/jobservice/acceptance/JobServiceEndToEndIT.java]: Why are you doing this and not using a `for loop` as I suggested? Any tests which have a delay set for the prerequisite jobs of over 5 seconds will fail using this method.

**Review by michael-bryson — CHANGES_REQUESTED** [review]
Make sure you put a proper title on this pull request.

File: job-service-acceptance-tests/src/test/java/com/hpe/caf/jobservice/acceptance/JobServiceEndToEndIT.java (hunk 2 above)
Comment (pintusingh135) [reply job-service-acceptance-tests/src/test/java/com/hpe/caf/jobservice/acceptance/JobServiceEndToEndIT.java]: Added waitUntilJobCompletes() method for job1Id

File: job-service-acceptance-tests/src/test/java/com/hpe/caf/jobservice/acceptance/JobServiceEndToEndIT.java (hunk 3 above)
Comment (pintusingh135) [reply job-service-acceptance-tests/src/test/java/com/hpe/caf/jobservice/acceptance/JobServiceEndToEndIT.java:356]: Added waitUntilJobCompletes() method for job1Id

File: job-service-acceptance-tests/src/test/java/com/hpe/caf/jobservice/acceptance/JobServiceEndToEndIT.java (hunk 4 above)
Comment (pintusingh135) [reply job-service-acceptance-tests/src/test/java/com/hpe/caf/jobservice/acceptance/JobServiceEndToEndIT.java]: Replaced the While loop with a for loop

**Review by michael-bryson — APPROVED** [review]

**Pull request merged**